implementations/rust/ockam/ockam_api/src/nodes/models/portal.rs

```diff
@@ -131,12 +131,12 @@ impl<'a> CreateOutlet<'a> {
 pub struct InletStatus<'a> {
     #[cfg(feature = "tag")]
     #[n(0)] tag: TypeTag<9302588>,
-    #[b(1)] pub bind_addr: Cow<'a, str>,
-    #[b(2)] pub worker_addr: Cow<'a, str>,
-    #[b(3)] pub alias: Cow<'a, str>,
+    #[n(1)] pub bind_addr: Cow<'a, str>,
```

Comment:
Types with lifetimes should have the `b` attribute so they can borrow values instead of copying them when decoding.

Also, we could replace `Cow<'a, str>` with `CowStr<'a>`, which is just a wrapper we introduced to enforce borrowing.

Reply:
Hi @adrianbenavides.

Thanks for the feedback. 

I've made changes (and pushed them to this PR), but now I get a compile error which has got me stuck (see below). Do you have any pointers that might help?

PS. The previous `InletList` and `OutletList` were based on what I saw for `ServiceList` and `TransportList`.

Thanks in advance 😄 

```
warning: /Users/neil/dev/ockam/Cargo.toml: unused manifest key: ockam_command
   Compiling ockam_api v0.19.0 (/Users/neil/dev/ockam/implementations/rust/ockam/ockam_api)
error: lifetime may not live long enough
   --> implementations/rust/ockam/ockam_api/src/nodes/models/portal.rs:218:24
    |
218 | #[derive(Debug, Clone, Decode, Encode)]
    |                        ^^^^^^
    |                        |
    |                        lifetime `'bytes` defined here
    |                        type annotation requires that `'bytes` must outlive `'a`
...
221 | pub struct InletList<'a> {
    |                      -- lifetime `'a` defined here
    |
    = help: consider adding the following bound: `'bytes: 'a`
    = note: this error originates in the derive macro `Decode` (in Nightly builds, run with -Z macro-backtrace for more info)

error: lifetime may not live long enough
   --> implementations/rust/ockam/ockam_api/src/nodes/models/portal.rs:238:24
    |
238 | #[derive(Debug, Clone, Decode, Encode)]
    |                        ^^^^^^
    |                        |
    |                        lifetime `'bytes` defined here
    |                        type annotation requires that `'bytes` must outlive `'a`
...
241 | pub struct OutletList<'a> {
    |                       -- lifetime `'a` defined here
    |
    = help: consider adding the following bound: `'bytes: 'a`
    = note: this error originates in the derive macro `Decode` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `ockam_api` due to 2 previous errors
```

Reply:
I changed the structs to the `b` attribute (and force pushed them to this PR). Now, `Rpc` can't be reused for the Inlet and Outlet requests.

```
error[E0502]: cannot borrow `*rpc` as mutable because it is also borrowed as immutable
   --> implementations/rust/ockam/ockam_command/src/node/show.rs:174:9
    |
171 |         let inlets = rpc.parse_response::<InletList>()?;
    |                      --------------------------------- immutable borrow occurs here
...
174 |         rpc.request(api::list_outlets()).await?;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
...
185 |             Some((&inlets, &outlets)),
    |                   ------- immutable borrow later used here

For more information about this error, try `rustc --explain E0502`.
```

https://github.com/build-trust/ockam/blob/8f364c0f33c658114f5c2ddc40543ce452121a18/implementations/rust/ockam/ockam_command/src/node/show.rs#L169-L186

The previous `InletList` and `OutletList` were based on what I see for `ServiceList` and `TransportList`.

Would you like me to __(1)__ change `print_query_status()` so it stops re-using `Rpc` between the Inlet and Outlet requests? or __(2)__ roll back so `InletList` and `OutletList` are like `ServiceList` and `TransportList` and do not borrow values?

Thanks in advance! 😄

Reply:
We can roll back to clone values and tackle this in a separate PR. Otherwise, this PR can get too big. Also, we have other structures that don't have the correct CBOR attributes, so it might be a good idea to step back and plan this carefully.

Sorry for the confusion!

Reply:
> Sorry for the confusion!

That's totally fine, this is all good experience for me.

In this last force push, I've kept the borrowing CBOR fields and implemented `Clone` for `Rpc`. Let me know what you think. If you'd prefer I roll back to the non-borrowing CBOR fields.

Thanks!

Reply:
That's perfect! It's what I had in mind since we don't care if we clone the responses on the CLI side. To be honest, I didn't think it would be that easy :laughing: